### PR TITLE
Reimplement "Improve how initial infections are treated in sid." (#38)

### LINKED ISSUE #41: Remove mandatory assort_by for initial conditions.
### What would you like to enhance and why? Is it related to an issue/problem?

Currently, the initial conditions require at least one assort by variable which ensures that initial infections spread in the population while keeping the shares of infections between assort by groups the same.

Problems:
- ``"county"`` is the default assort by variable which is not used by everybody.
- an assort by variable is necessary.
- Also requires known_cases_multiplier

### PR DESCRIPTION
**Pull request by tobiasraabe**
Closes #41.

### CODE REVIEW

File: src/sid/simulate.py [hunk 1]
```diff
@@ -59,7 +61,10 @@ def get_simulate_func(
         initial_states (pandas.DataFrame): See :ref:`states`. Cannot contain the column
             "date" because it is used internally.
         initial_infections (pandas.Series): Series with the same index as states with
-            initial infections.
+            initial infections. They are initial known cases. Sid will assume that
```
Comment (janosg) [inline src/sid/simulate.py:64]: I think as it is now, the `initial_infections` should be part of the `initial_conditions` dictionary.

**Review by janosg — APPROVED** [review]
If I see it correctly, after this PR it is still not possible to start the simulation with the final states of a previous simulation as initial states. Do you want to postpone that until simulate gets a better results object? The rest is very good. Just had one comment.

File: src/sid/simulate.py (hunk 1 above)
Comment (tobiasraabe) [reply src/sid/simulate.py:64]: Make sense. It is not necessary if we start the simulation from a previous states DataFrame.

**Comment by tobiasraabe** [comment]
You are right. I postponed the extraction to make it in another PR. The changes were already enough.

**Review by janosg — APPROVED** [review]

**Pull request merged**